pkg/deploy/registry/deployconfig/strategy.go

```diff
@@ -39,14 +41,53 @@ func (strategy) AllowUnconditionalUpdate() bool {
 
 // PrepareForCreate clears fields that are not allowed to be set by end users on creation.
 func (strategy) PrepareForCreate(obj runtime.Object) {
-	_ = obj.(*api.DeploymentConfig)
-	// TODO: need to ensure status.latestVersion is not set out of order
+	dc := obj.(*api.DeploymentConfig)
+	dc.Generation = 1
+	dc.Status = api.DeploymentConfigStatus{}
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (strategy) PrepareForUpdate(obj, old runtime.Object) {
-	_ = obj.(*api.DeploymentConfig)
-	// TODO: need to ensure status.latestVersion is not set out of order
+	newDc := obj.(*api.DeploymentConfig)
+	oldDc := old.(*api.DeploymentConfig)
+
+	newVersion := newDc.Status.LatestVersion
+	oldVersion := oldDc.Status.LatestVersion
+
+	// Check for imagechange controller change
+	isImageControllerChange := deployutil.IsImageChangeControllerChange(*newDc, *oldDc)
+
+	// Persist status
+	newStatus := newDc.Status
+	newDc.Status = oldDc.Status
+
+	// If this update comes from the imagechange controller, tolerate status detail updates.
```

Comment:
Yes, I will address this in https://github.com/openshift/origin/pull/8746 by removing the status update. So the flow on an image change will be: ImageChange controller updates spec ---(watch triggered)---> ConfigController sees a template diff, updates latestVersion ---(watch triggered)---> main dc controller runs the new deployment.